m3/contract/filecoin.mock.sol

```diff
@@ -0,0 +1,671 @@
+        bytes aggregate_proof;
+    }
+    struct ProveReplicaUpdatesParams {
+        CommonTypes.ReplicaUpdate[] updates;
+    }
+    struct ProveReplicaUpdatesParams2 {
+        CommonTypes.ReplicaUpdate2 updates;
+    }
+    struct DisputeWindowedPoStParams {
+        uint64 deadline;
+        uint64 post_index;
+    }
+    struct PreCommitSectorParams {
+        CommonTypes.RegisteredSealProof seal_proof;
+        uint64 sector_number;
+        CommonTypes.CID sealed_cid;
+        int64 seal_rand_epoch;
+        uint64[] deal_ids;
+        int64 expiration;
+        bool replace_capacity;
+        uint64 replace_sector_deadline;
+        uint64 replace_sector_partition;
+        uint64 replace_sector_number;
+    }
+
+    struct PreCommitSectorBatchParams {
+        PreCommitSectorParams[] sectors;
+    }
+
+    struct PreCommitSectorBatchParams2 {
+        CommonTypes.SectorPreCommitInfo[] sectors;
+    }
+
+    struct ProveCommitSectorParams {
+        uint64 sector_number;
+        bytes proof;
+    }
+
+    struct ConfirmSectorProofsParams {
+        uint64[] sectors;
+        CommonTypes.FilterEstimate reward_smoothed;
+        int256 reward_baseline_power;
+        CommonTypes.FilterEstimate quality_adj_power_smoothed;
+    }
+    struct CheckSectorProvenParams {
+        uint64 sector_number;
+    }
+    struct ExtendSectorExpirationParams {
+        CommonTypes.ExpirationExtension[] extensions;
+    }
+
+    struct ExtendSectorExpiration2Params {
+        CommonTypes.ExpirationExtension2[] extensions;
+    }
+
+    struct TerminateSectorsParams {
+        CommonTypes.TerminationDeclaration[] terminations;
+    }
+
+    struct TerminateSectorsReturn {
+        bool done;
+    }
+
+    struct DeclareFaultsParams {
+        CommonTypes.FaultDeclaration[] faults;
+    }
+
+    struct DeclareFaultsRecoveredParams {
+        CommonTypes.RecoveryDeclaration[] recoveries;
+    }
+
+    struct CompactPartitionsParams {
+        uint64 deadline;
+        uint8 partitions;
+    }
+
+    struct CompactSectorNumbersParams {
+        uint8 mask_sector_numbers;
+    }
+
+    struct ApplyRewardParams {
+        int256 reward;
+        int256 penalty;
+    }
+
+    struct ReportConsensusFaultParams {
+        bytes header1;
+        bytes header2;
+        bytes header_extra;
+    }
+
+    struct WithdrawBalanceParams {
+        int256 amount_requested;
+    }
+
+    struct WithdrawBalanceReturn {
+        int256 amount_withdrawn;
+    }
+
+    struct ChangeBeneficiaryParams {
+        CommonTypes.Addr new_beneficiary;
+        int256 new_quota;
+        uint64 new_expiration;
+    }
+
+    struct GetBeneficiaryReturn {
+        CommonTypes.ActiveBeneficiary active;
+        CommonTypes.PendingBeneficiaryChange proposed;
+    }
+
+    struct DeferredCronEventParams {
+        bytes event_payload;
+        CommonTypes.FilterEstimate reward_smoothed;
+        CommonTypes.FilterEstimate quality_adj_power_smoothed;
+    }
+}
+
+library MinerAPI{
```

Comment:
These methods are invoked on a particular address. Usually in Solidity you would instantiate a proxy to a contract by specifying its address: https://medium.com/@blockchain101/calling-the-function-of-another-contract-in-solidity-f9edfa921f4c But I'm not sure using that pattern is possible here because Solidity assumes it is an Ethereum cross-contract call, whereas in this case we want to treat the target address as an instance variable.